doc/specs/application-auth-token.md

```diff
@@ -20,26 +35,80 @@ A semver string specifying the spec version under which this AAT needs to be int
 
 > type: `string`
 
-The application will sign a hash of the `message` property within this token with the specified `appPubKey` and
-corresponding private key.
+The `Application` will sign a hash of a marshalled AAT structure where all the fields
+are populated with the exception of the `Signature` itself. The resultant `Signature`
+is the one that's populated in the `AAT`.
 
 ### applicationPublicKey
 
 > type: `string`
 
-The hexadecimal publicKey of the Application
+The hexadecimal `publicKey` of the `Application`
 
 ### clientPublicKey
 
 > type: `string`
 
-Required for signature verification, the hexadecimal public of each individual client allowing for granular control of
-who can use the AAT
+Required for signature verification.
+
+Also known as the `gatewayPublicKey`.
+
+The hexadecimal public key allowing granular control of who can use the `AAT`.
 
 ## ECDSA ed25519 Signature Scheme
 
-The protocol wide ed25519 ECDSA will be used for any signatures and verifications that are used within this
-specification.
+The protocol wide ed25519 ECDSA will be used for any signatures and verifications
+that are used within this specification.
+
+## AAT Generation
+
+When generating a new AAT, the owner of the `Application` private key has two options:
+
+1. Use the same public key for signing the AAT and the relay requests
```

Comment:
If it's "When generating a new AAT", what the application owner can choose is which public key to be set as clientPublicKey.